airflow/contrib/hooks/azure_batchai_hook.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+
+from airflow.hooks.base_hook import BaseHook
+from airflow.exceptions import AirflowException
+
+from azure.common.client_factory import get_client_from_auth_file
+from azure.common.credentials import ServicePrincipalCredentials
+
+from azure.mgmt.batchai import BatchAIManagementClient
+
+
+class AzureBatchAIHook(BaseHook):
+    """
+    Interact with Azure Batch AI
+    :param azure_batchai_conn_id: Reference to the Azure Batch AI connection
+    :type azure_batchai_conn_id: str
+    :param config_data: JSON Object with credential and subscription information
+    :type config_data: str
+    """
+
+    def __init__(self, azure_batchai_conn_id='azure_batchai_default', config_data=None):
+        self.conn_id = azure_batchai_conn_id
+        self.connection = self.get_conn()
+        self.configData = config_data
+        self.credentials = None
+        self.subscription_id = None
+
+    def get_conn(self):
+        try:
+            conn = self.get_connection(self.conn_id)
+            key_path = conn.extra_dejson.get('key_path', False)
+            if key_path:
+                if key_path.endswith('.json'):
+                    self.log.info('Getting connection using a JSON key file.')
+                    return get_client_from_auth_file(BatchAIManagementClient,
+                                                     key_path)
+                else:
+                    raise AirflowException('Unrecognised extension for key file.')
+
+            elif os.environ.get('AZURE_AUTH_LOCATION'):
```

Comment:
Why we need this? Airflow currently supports the creation of connection using environment variables. https://airflow.apache.org/howto/manage-connections.html#creating-a-connection-with-environment-variables

Reply:
This is because a user may want to use Azure AD to store their credentials

Reply:
The user can define a new connection using the environment variable `AIRFLOW_CONN_*`. For example: `AIRFLOW_CONN_AZURE_DEFAULT`. Official documentation is missing, but you can look at this PR: https://github.com/apache/airflow/pull/4523/files
I found another operator who has a very similar looking code, so if it really has to be it should not duplicate the code that is already in the project.
https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/azure_container_instance_hook.py